meson.build

```diff
@@ -1611,6 +1611,7 @@ fuzzers = []
 
 basic_includes = include_directories(
         'src/basic',
+        'src/firmware-common',
```

Comment:
A bit of a bikeshed comment, but maybe we could call this something different? I think even `src/common` would be nicer. Or `src/generic`.  `firmware-common` is lengthy and `sd-boot` isn't actually firmware. (EFI is "firmware *interface*. I always thought we're on the non-firmware side of that boundary.)

Reply:
I'd prefer "generic" or "common". But "fundamental" works too.